MarioGame/Source/Systems/FireBoolSystem.cs

```diff
@@ -0,0 +1,284 @@
+                {
+                    fireballsToMove.Add(fireball);
+                }
+            }
+
+            foreach (var fireball in fireballsToMove)
+            {
+                MoveFireballAfterWait(fireball);
+                fireballTimers.Remove(fireball);
+            }
+        }
+    }
+
+    private static void createFire(IEnumerable<Entity> entities, float positionX, float positionY, List<Action> pendingActions, CameraComponent cameraComponent, float forwardSpeedFire)
+    {
+        IEnumerable<Entity> fireballEntities = entities
+            .WithComponents(typeof(FireBoolComponent), typeof(AnimationComponent), typeof(ColliderComponent))
+            .Where(entity => entity.GetComponent<ColliderComponent>().collider.BodyType == BodyType.Static);
+
+        var firstFireball = fireballEntities.FirstOrDefault();
+
+        if (firstFireball != null)
+        {
+            var collider = firstFireball.GetComponent<ColliderComponent>();
+            var fireballComponent = firstFireball.GetComponent<FireBoolComponent>();
+
+            collider.collider.Position = new AetherVector2(positionX, positionY);
+            collider.collider.BodyType = BodyType.Dynamic;
+            collider.collider.OnCollision += (fixtureA, fixtureB, contact) => HandleCollision(fixtureA, fixtureB, contact, entities, pendingActions, cameraComponent, forwardSpeedFire);
+            collider.collider.LinearVelocity = new AetherVector2(forwardSpeedFire, 0);
+
+            fireballComponent.InitialDirection = Math.Sign(forwardSpeedFire);
+        }
+    }
+
+
+private static bool HandleCollision(Fixture fixtureA, Fixture fixtureB, Contact contact, IEnumerable<Entity> entities, List<Action> pendingActions,CameraComponent cameraComponent,float forwardSpeedFire)
+{
+    Entity fireballEntity = entities.FirstOrDefault(e =>
+        e.GetComponent<ColliderComponent>()?.collider == fixtureA.Body ||
+        e.GetComponent<ColliderComponent>()?.collider == fixtureB.Body);
+
+    if (fireballEntity == null) return true;
+
+    if (fireballEntity.GetComponent<FireBoolComponent>() == null)
+    {
+        return true;
+    }
+    var fireballComponent = fireballEntity.GetComponent<FireBoolComponent>();
+    forwardSpeedFire = fireballComponent.InitialDirection * Math.Abs(forwardSpeedFire);
+
+    Entity otherEntity = entities.FirstOrDefault(e =>
+        e.GetComponent<ColliderComponent>()?.collider ==
+        (fixtureA.Body == fireballEntity.GetComponent<ColliderComponent>()?.collider ? fixtureB.Body : fixtureA.Body));
+    if (otherEntity != null)
+    {
+        var animationComponent = otherEntity.GetComponent<AnimationComponent>();
+        if (animationComponent != null)
+        {
+            string otherEntityName = Animations.entityTextures
+                .FirstOrDefault(x => x.Value == animationComponent.animations)
+                .Key.ToString() ?? string.Empty;
+
+            if (otherEntityName == "GOOMBA" || otherEntityName == "KOOPA")
```

Comment:
You can use the collision categories to simplify the logic of handling collisions with enemies, i.e. set up that fireball colliders can only collide with colliders of enemy type, review the property CollistionWith here:
https://nkast.github.io/Aether.Physics2D/Documentation/2.0/classnkast_1_1_aether_1_1_physics2_d_1_1_dynamics_1_1_fixture.html